### DISCUSSION TITLE
Feature variations important; unusable for many real projects without

### DISCUSSION THREAD

**Discussion by tphinney** [adobe-type-tools/feature_file_change_review#11] [General]
I am sure not all users have the same needs I do. But I can say that both of the real variable font projects I have recently been involved with use _feature variations_.

One (Google Symbols / Material Symbols) uses them for about half the characters in the font, and the other (Science Gothic) uses it for just a handful of showcase characters. But even then, it is important for the dollar sign to… not fail in some parts of the design space.

**Comment by skef** [comment]
As noted in the ReadMe we hope to address feature variations in the first half of next year. 

We know that this work is happening far later than expected. We decided to split the review into two stages so that implementations could start integrating the variable value changes while work was continuing on the Feature Variations changes.

**Reply by jmsole** [reply]
@skef Has there been any progress with this? It would be great to have something solid and ready to be implemented without fear of things changing.

**Comment by skef** [comment]
I'm working on getting the process described toward the end of https://github.com/fonttools/fonttools/issues/3953 started up. I have also been looking at the problem again.

I haven't discussed this widely but a good part of the delay is just that the combination of upcoming features in ISO Open Font Format draft form and the current terrible specification of feature variations makes a good solution that can work with both the current and the expected complicated. The problem isn't just specifying the grammar, it's providing a robust enough reference implementation that other feature file compilers will actually pick up and use the grammar. I'm hoping the meetings will help with that.
